gridstatus/isone.py

```diff
@@ -106,8 +106,30 @@ def get_status(self, date, verbose=False):
     #     mix_df.columns.name = None
     #     return mix_df
 
+    def _assign_dst_aware_time(
+        self,
+        series: pd.Series,
+        transition_idx: int | None,
+    ) -> pd.Series:
+        if transition_idx is None:
+            return series.dt.tz_localize("US/Eastern", ambiguous="infer")
+
+        dates = pd.Series(index=series.index, dtype="datetime64[ns, US/Eastern]")
+        dates[series.index < transition_idx] = series[
```

Comment:
you can [pass a boolean mask](https://pandas.pydata.org/docs/reference/api/pandas.Series.tz_localize.html#:~:text=bool%2Dndarray%20where%20True%20signifies%20a%20DST%20time%2C%20False%20designates%20a%20non%2DDST%20time%20(note%20that%20this%20flag%20is%20only%20applicable%20for%20ambiguous%20times)) to `ambiguous`

would this be the same thing? 

`data.dt.tz_localize("US/Eastern", ambiguous=series.index < transition_idx)`